hedera-node/hedera-app/src/main/java/com/hedera/node/app/Hedera.java

```diff
@@ -435,26 +443,33 @@ public void init(@NonNull final Platform platform, @NonNull final NodeId nodeId)
             // server when we fall behind or ISS.
             final var notifications = platform.getNotificationEngine();
             notifications.register(PlatformStatusChangeListener.class, notification -> {
+                final var wasActive = platformStatus == PlatformStatus.ACTIVE;
                 platformStatus = notification.getNewStatus();
                 switch (notification.getNewStatus()) {
-                    case ACTIVE -> logger.info("Hederanode#{} is ACTIVE", nodeId);
+                    case ACTIVE -> {
+                        logger.info("Hederanode#{} is ACTIVE", nodeId);
+                        startGrpcServer();
+                    }
                     case BEHIND -> {
                         logger.info("Hederanode#{} is BEHIND", nodeId);
-                        shutdownGrpcServer();
-                    }
-                    case FREEZE_COMPLETE -> {
-                        logger.info("Hederanode#{} is in FREEZE_COMPLETE", nodeId);
-                        shutdownGrpcServer();
+                        if (wasActive) shutdownGrpcServer();
                     }
+                    case FREEZE_COMPLETE -> logger.info("Hederanode#{} is in FREEZE_COMPLETE", nodeId);
                     case REPLAYING_EVENTS -> logger.info("Hederanode#{} is REPLAYING_EVENTS", nodeId);
                     case STARTING_UP -> logger.info("Hederanode#{} is STARTING_UP", nodeId);
                     case CATASTROPHIC_FAILURE -> {
                         logger.info("Hederanode#{} is in CATASTROPHIC_FAILURE", nodeId);
-                        shutdownGrpcServer();
+                        if (wasActive) shutdownGrpcServer();
+                    }
+                    case CHECKING -> {
+                        logger.info("Hederanode#{} is CHECKING", nodeId);
+                        if (wasActive) shutdownGrpcServer();
                     }
-                    case CHECKING -> logger.info("Hederanode#{} is CHECKING", nodeId);
                     case OBSERVING -> logger.info("Hederanode#{} is OBSERVING", nodeId);
-                    case FREEZING -> logger.info("Hederanode#{} is FREEZING", nodeId);
+                    case FREEZING -> {
+                        logger.info("Hederanode#{} is FREEZING", nodeId);
+                        if (wasActive) shutdownGrpcServer();
+                    }
```

Comment:
Basic cleanup -- On each transaction from/to active, we have to shutdown/startup the grpc server.